src/main/java/core/basesyntax/service/strategy/strategyimpl/BalanceOperation.java

```diff
@@ -0,0 +1,11 @@
+package core.basesyntax.service.strategy.strategyimpl;
+
+import core.basesyntax.dao.Storage;
+import core.basesyntax.model.FruitRecord;
+
+public class BalanceOperation implements OperationHandler {
+    @Override
+    public void apply(FruitRecord transaction) {
+        Storage.storage.put(transaction.getFruit(), transaction.getQuantity());
```

Comment:
Consider checking if 'transaction.getQuantity()' is non-negative before putting it into the storage. While the 'BalanceOperation' is meant to set the initial balance and it is unlikely to be negative, it is a good practice to validate the input data to ensure the integrity of the storage state.